### PR TITLE
[Batch] Updates for new extension model

### PR DESCRIPTION
**Pull request by annatisch**
This PR includes changes to the Batch module to support the new extension model:
- Removes imports for old extension model
- Includes azure-namespace dependencies required by new extension package

---
### General Guidelines

- [x] The PR has modified HISTORY.rst describing any customer-facing, functional changes. Note that this does not include changes only to help content. (see [Modifying change log](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules#modify-change-log)).

### CODE REVIEW

**Comment by annatisch** [comment]
FYI @xingwu1 @matthchr

File: src/command_modules/azure-cli-batch/azure/cli/command_modules/batch/__init__.py [hunk 1]
```diff
@@ -12,19 +12,7 @@
...
-    try:
-        import azure.cli.command_modules.batch_extensions._params
-    except ImportError as exp:
-        # Optional Batch Extensions not installed
-        logger.debug("Did not load Batch Extensions module")
-        logger.debug(str(exp))
 
 
 def load_commands():
     import azure.cli.command_modules.batch.commands  # pylint: disable=redefined-outer-name, unused-variable
```
Comment (annatisch) [inline src/command_modules/azure-cli-batch/azure/cli/command_modules/batch/__init__.py]: @derekbekoe - I'm correct in assuming that the new extension model no longer requires these imports, right? :)

Comment (derekbekoe) [reply src/command_modules/azure-cli-batch/azure/cli/command_modules/batch/__init__.py]: Correct

File: src/command_modules/azure-cli-batch/setup.py [hunk 2]
```diff
@@ -34,6 +34,7 @@
 
 DEPENDENCIES = [
     'azure-batch==4.0.0',
+    'azure-batch-extensions>=1.0,<1.1',
```
Comment (derekbekoe) [inline src/command_modules/azure-cli-batch/setup.py]: What's this for?
I guess it has something to do with this comment "Includes azure-namespace dependencies required by new extension package"?

Comment (annatisch) [reply src/command_modules/azure-cli-batch/setup.py]: @derekbekoe - Yes this is our SDK extensions package. It uses the azure namespace when installing so we cannot add it as a dependency to the new CLI extension wheel, so we have to add it here instead in order for it to be installed correctly.

Comment (derekbekoe) [reply src/command_modules/azure-cli-batch/setup.py]: Okay yes.

**Pull request merged**